### DISCUSSION TITLE
Mid Lesson test challenge.

### DISCUSSION THREAD

**Discussion by santiagoib** [Cyfrin/foundry-full-course-cu#1150] [Show and tell]
Gm, topgs, this is the way I did the test, but I'm pretty sure there are ways to do it cleaner and more efficient. But this way it was easy for me to structure it.
```solidity
function testCheckUpkeepReturnsFalseIfEnoughTimeHasntPassed() public {
        // Arrange
        vm.warp(160 seconds); // set the time stamp
        uint256 lastTimeStamp = 100 seconds; // set the last time stamp
        uint256 interval = 1 minutes; // set the interval 
        uint256 diff = block.timestamp - lastTimeStamp; // i do this to put it into the parameters 
        
        //  Act/Assert
        assertLe(diff, interval);  // and i finally assert
    }
```
So the main idea is to check if enough time has passed and if not it would revert.
Tell me what do u think, gl.

**Comment by 0xusmanf** [comment]
I am not sure if you left that part intentionally but you are not interacting at all with the contract function you are trying to test.
